bpf/tests/ipsec_from_lxc_tunnel.c

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+#include "common.h"
+#include <bpf/ctx/skb.h>
+#include "pktgen.h"
+#define ROUTER_IP
+#include "config_replacement.h"
+#undef ROUTER_IP
+
+#define TUNNEL_MODE
+#define HAVE_ENCAP
+#define ENABLE_ROUTING
```

Comment:
Yes, but I handled it wrong. Let it rectify.

In short, `ENABLE_ROUTING` means the endpoint requires BPF routing instead of stack routing. When endpointRoutes is enabled, the macro doesn't exist.

In details, config.RequireRouting = false if option.Config.EnableEndpointRoutes:
https://github.com/cilium/cilium/blob/28a3cb7077aec3b03e54b873765cb764b0d46fa0/pkg/endpoint/endpoint.go#L965-L978
Then cilium skips `#define ENABLE_ROUTING 1`:
https://github.com/cilium/cilium/blob/28a3cb7077aec3b03e54b873765cb764b0d46fa0/pkg/datapath/linux/config/config.go#L1081-L1082

Back to my test cases, only ipsec_from_*_endpointroutes.c should NOT have this macro, while ipsec_from_nativerouting.c should have it, but I messed it up.

I'll also double check other macro definition differences among combinations.